ctapipe/tools/stage1.py

```diff
@@ -0,0 +1,767 @@
+            )
+
+        # setup HDF5 compression:
+        self._hdf5_filters = tables.Filters(
+            complevel=self.compression_level,
+            complib=self.compression_type,
+            fletcher32=True,  # attach a checksum to each chunk for error correction
+        )
+
+    def _write_simulation_configuration(self, writer, event):
+        """
+        Write the simulation headers to a single row of a table. Later 
+        if this file is merged with others, that table will grow. 
+
+        Note that this function should be run first
+        """
+        self.log.debug("Writing simulation configuration")
+
+        class ExtraMCInfo(Container):
+            container_prefix = ""
+            obs_id = Field(0, "MC Run Identifier")
+
+        extramc = ExtraMCInfo()
+        extramc.obs_id = event.index.obs_id
+        event.mcheader.prefix = ""
+        writer.write("configuration/simulation/run_config", [extramc, event.mcheader])
+
+    def _write_simulation_histograms(self, writer: HDF5TableWriter):
+        """ Write the distribution of thrown showers
+
+        Notes
+        -----
+        - this only runs if this is a simulation file. The current implementation is
+          a bit of a hack and implies we should improve SimTelEventSource to read this
+          info.
+        - Currently the histograms are at the end of the simtel file, so if max_events
+          is set to non-zero, the end of the file may not be read, and this no
+          histograms will be found.
+        """
+        self.log.debug("Writing simulation histograms")
+
+        def fill_from_simtel(
+            obs_id, eventio_hist, container: SimulatedShowerDistribution
+        ):
+            """ fill from a SimTel Histogram entry"""
+            container.obs_id = obs_id
+            container.hist_id = eventio_hist["id"]
+            container.num_entries = eventio_hist["entries"]
+            xbins = np.linspace(
+                eventio_hist["lower_x"],
+                eventio_hist["upper_x"],
+                eventio_hist["n_bins_x"] + 1,
+            )
+            ybins = np.linspace(
+                eventio_hist["lower_y"],
+                eventio_hist["upper_y"],
+                eventio_hist["n_bins_y"] + 1,
+            )
+
+            container.bins_core_dist = xbins * u.m
+            container.bins_energy = 10 ** ybins * u.TeV
+            container.histogram = eventio_hist["data"]
+            container.meta["hist_title"] = eventio_hist["title"]
+            container.meta["x_label"] = "Log10 E (TeV)"
+            container.meta["y_label"] = "3D Core Distance (m)"
+
+        if type(self.event_source) is not SimTelEventSource:
+            return
+
+        hists = self.event_source.file_.histograms
+        if hists is not None:
+            hist_container = SimulatedShowerDistribution()
+            hist_container.prefix = ""
+            for hist in hists:
+                if hist["id"] == 6:
```

Comment:
Why only the 6th histogram? At least simulated and triggered events would be interesting, right?

Reply:
this is simulated events, needed for the computation of effective areas.  We could of course store others, if there is a use case for them.  Even this can now be re-computed from the data (since we keep triggered and simulated event info).